File: benchmarks/call-closure-mono/bench.lua
local function makeAdder(base)
  return function(x)
    return x + base
  end
end

local iterations = 20000000
local f = makeAdder(7)
local acc = 0

for i = 0, iterations - 1 do
  acc = acc + f(i % 1024)
end

print(acc)

File: benchmarks/call-closure-mono/bench.js
function makeAdder(base) {
  return function (x) {
    return x + base;
  };
}

function main() {
  const iterations = 20000000;
  const f = makeAdder(7);
  let acc = 0;

  for (let i = 0; i < iterations; i++) {
    acc += f(i & 1023);
  }

  console.log(acc);
}

main();
